src/main/scala/rocket/ABLU.scala

```diff
@@ -0,0 +1,278 @@
+  // also BCLR
+  val and = adder_in1 & logic_in2
+  // also BSET
+  val or = adder_in1 | logic_in2
+  val bext = and.orR
+
+  // SLT, SLTU
+  // BEQ, BNE, BLT, BGE
+  // MAX, MIN
+  val slt =
+    Mux(io.in1(xLen-1) === io.in2(xLen-1), adder_out(xLen-1),
+    Mux(isUnsigned, io.in2(xLen-1), io.in1(xLen-1)))
+  val cmp = isInverted ^ Mux(isSEQSNE, ~(xor.orR), slt)
+  io.cmp_out := cmp
+  // MAX, MAXU, MIN, MINU
+  val max_min = Mux(cmp, io.in2, io.in1)
+
+  // counter
+  // CLZ, CPOP, CTZ
+  val cpop = PopCount(
+    if (xLen == 32) io.in1
+    else {
+      require(xLen == 64)
+      Mux(io.dw === DW_64, io.in1, Cat(Fill(32, 0.U(1.W)), io.in1(31,0)))
+    })
+  // ctz_in = ~adder_out & adder_in1 // all zero or one hot
+  val ctz_in = and
+  val ctz_out = Cat(~ctz_in.orR, VecInit((0 to log2Ceil(xLen)-1).map(
+    x => {
+      val bits = ctz_in.asBools.zipWithIndex
+      VecInit(
+        bits
+          filter { case (_, i) => i % (1 << (x + 1)) >= (1 << x) }
+          map { case (b, _) => b }
+        ).asUInt.orR
+    }
+  ).toSeq).asUInt)
+
+  // ZEXT/SEXT
+  val exth = Cat(Fill(xLen-16, Mux(isSEXT, io.in1(15), 0.U)), io.in1(15,0))
+  val extb = Cat(Fill(xLen-8, io.in1(7)), io.in1(7,0))
+
+  // REV/ORC
+  def asBytes(in: UInt): Vec[UInt] = VecInit(in.asBools.grouped(8).map(VecInit(_).asUInt).toSeq)
+  val in1_bytes = asBytes(io.in1)
+  val rev8 = VecInit(in1_bytes.reverse.toSeq).asUInt
+  val orc_brev8 = VecInit(in1_bytes.map(x => {
+    val orc = Mux(x.orR, 0xFF.U(8.W), 0.U(8.W))
+    // BREV8 only in Zbk
+    if (usingBitManipCrypto)
+      Mux(isORC, orc, Reverse(x))
+    else orc
+  }).toSeq).asUInt
+
+  // pack
+  def sext(in: UInt): UInt = {
+    val in_hi_32 = Fill(32, in(31))
+    Cat(in_hi_32, in)
+  }
+  val pack = if (usingBitManipCrypto) {
+    if (xLen == 32) Cat(io.in2(xLen/2-1,0), io.in1(xLen/2-1,0))
+    else {
+      require(xLen == 64)
+      Mux(io.dw === DW_64,
+        Cat(io.in2(xLen/2-1,0), io.in1(xLen/2-1,0)),
+        sext(Cat(io.in2(xLen/4-1,0), io.in1(xLen/4-1,0))))
+    }
+  } else 0.U
+  val packh = if (usingBitManipCrypto) Cat(0.U((xLen-16).W), io.in2(7,0), io.in1(7,0)) else 0.U
+
+  // zip
+  val zip = if (xLen == 32 && usingBitManipCrypto) {
+    val lo = io.in1(15,0).asBools
+    val hi = io.in1(31,16).asBools
+    VecInit(lo.zip(hi).map { case (l, h) => VecInit(Seq(l, h)).asUInt }).asUInt
+  } else 0.U
+  val unzip = if (xLen == 32 && usingBitManipCrypto) {
+    val bits = io.in1.asBools.zipWithIndex
+    val lo = VecInit(bits filter { case (_, i) => i % 2 == 0 } map { case (b, _) => b }).asUInt
+    val hi = VecInit(bits filter { case (_, i) => i % 2 != 0 } map { case (b, _) => b }).asUInt
+    Cat(hi, lo)
+  } else 0.U
+
+  val out = Mux1H(out1H, Seq(
+    adder_out,
+    shro,
+    and,
+    xor,
+    //
+    or,
+    bext,
+    cmp,
+    max_min,
+    //
+    cpop,
+    ctz_out,
+    exth,
+    extb,
+    //
+    rev8,
+    orc_brev8,
+    pack,
+    packh,
+    //
+    zip,
+    unzip))
```

Comment:
Seems to be a high fanout `out`...
But I don't have so much idea here to provide better circuit performance...